utils.js

```diff
@@ -322,9 +322,16 @@ export class InjectionsHandler extends BasicHandler {
         if (!(original instanceof Function))
             throw new Error(`Virtual function ${name}() is not available for ${object}`);
 
-        object[name] = function (...args) {
-            return injectedFunction.call(this, original, ...args);
-        };
+        if (original.constructor.name === 'AsyncFunction') {
```

Comment:
Oh, funny thing... I had this changed months ago and I forgot to commit...

However, define something like `const AsyncFunctionType = (async () => {}).constructor;` and then check for `original instanceof AsyncFunctionType`.

Also I feel we should check if the two things match, so if both the original ones are async or not and depending on that behave differently.

Reply:
I saw that suggestion when googling but it felt like a hack so I intentionally didn't do it that way. It's uglier, more code, and not obviously correct to me.

Good point about checking two things match though...

Since it turns out the bug #2110 actually isn't important, and it's fixed by #2166, and you have another fix in mind, I suggest just merging your own fix or wait for #2166.